### ISSUE TITLE
Why i get double split line between directory and file ?

### ISSUE THREAD

**Issue by parkerhsu** [greggman/unzipit#27]
The original file structure is like this:

![origin_zip](https://user-images.githubusercontent.com/35717670/225505829-dd9b5662-1be3-4d9b-89c6-9adde9cc40f9.png)

while i unzip this file, the result is:

![WX20230316-111859@2x](https://user-images.githubusercontent.com/35717670/225505905-6f1cd8cc-590f-421d-8657-3020de8bb1a6.png)

**Comment by greggman** [comment]
My guess is that's the name in the file

**Comment by parkerhsu** [comment]
> the

I don't understand, is that you mean the "/" is contained in the filename but i can't see it?

**Comment by parkerhsu** [comment]
> My guess is that's the name in the file

I don't understand, is that you mean the "/" is contained in the filename but i can't see it?

**Comment by greggman** [comment]
did you make the file? Can you post it or make a new file with the same problem and post it ? Just drag it into the issue here, make sure there's no private info

**Comment by parkerhsu** [comment]
> did you make the file? Can you post it or make a new file with the same problem and post it ? Just drag it into the issue here, make sure there's no private info

Sorry，the original zip file came from company's platform, i can't show it here. I tried to unzip the file manually and zip it again, the wired thing is when i use unzipit to unzip the file zipped by myself , everything is all fine and double splash disappeared.

My current solution is filtered the double splash, i hope my problem can be useful.

**Comment by greggman** [comment]
Yes, that's super useful to know. I'll add a comment to the README.

It's unzipit's policy to make filename issues the responsibility of the caller. From unzipit's POV the names are just strings in the file. They could be anything including characters that are not compatible with your filesystem. For example if the name has `:` or `"` or `*` or `?` or  `<` or `>` they're not valid on Windows but I can easily make a file called `C:Foo` on Mac/Linux and then put it in a zip file. Unzipit it can't know all the limits so it just gives you the string from the file.

I'd never personally seen a file with `//` in it so thank you for the example.
